crates/settlement-clients/ethereum/src/tests/mod.rs

```diff
@@ -77,61 +73,83 @@ sol! {
 
-    // Setup Anvil
-    let anvil = Anvil::new()
-        .port(*PORT)
-        .fork(&*ETH_RPC)
-        .fork_block_number(block_no - 1)
-        .try_spawn()
-        .expect("Could not spawn Anvil.");
+        // Setup Anvil
+        let anvil = match self.fork_block {
+            Some(fork_block) => {
+                Anvil::new().fork(&*ETH_RPC).fork_block_number(fork_block).try_spawn().expect("Could not spawn Anvil.")
+            }
+            None => Anvil::new().try_spawn().expect("Could not spawn Anvil."),
+        };
 
-    // Setup Provider
-    let provider = ProviderBuilder::new().on_http(anvil.endpoint_url());
+        // Setup Provider
+        let provider = ProviderBuilder::new().on_http(anvil.endpoint_url());
 
-    // Setup EthereumSettlementClient
-    let settings_provider: DefaultSettingsProvider = DefaultSettingsProvider {};
-    let ethereum_settlement_client = EthereumSettlementClient::with_test_settings(&settings_provider, provider.clone());
+        if let Some(impersonator) = self.impersonator.clone() {
+            provider.anvil_impersonate_account(impersonator).await.expect("Unable to impersonate account.");
+        }
+
+        let rpc_url = anvil.endpoint_url();
 
-    TestSetup { anvil, ethereum_settlement_client, provider }
+        EthereumTest { _anvil: anvil, provider, rpc_url }
+    }
 }
 
 #[rstest]
 #[tokio::test]
-#[case::basic(20468828)]
 /// Tests if the method is able to do a transaction with same function selector on a dummy contract.
 /// If we impersonate starknet operator then we loose out on testing for validity of signature in the transaction.
 /// Starknet core contract has a modifier `onlyOperator` that restricts anyone but the operator to send transaction to `updateStateKzgDa` method
 /// And hence to test the signature and transaction via a dummy contract that has same function selector as `updateStateKzgDa`.
 /// and anvil is for testing on fork Eth.
-async fn update_state_blob_with_dummy_contract_works(#[case] fork_block_no: u64) {
-    env::set_var("SHOULD_IMPERSONATE_ACCOUNT", "false");
-    let TestSetup { anvil, ethereum_settlement_client, provider } = setup_ethereum_test(fork_block_no);
+async fn update_state_blob_with_dummy_contract_works() {
+    let setup = EthereumTestBuilder::new().build().await;
 
-    println!("{:?}", anvil);
     // Deploying a dummy contract
-    let contract = DummyCoreContract::deploy(&provider).await.expect("Unable to deploy address");
-    assert_eq!(
-        contract.address().to_string(),
-        *TEST_DUMMY_CONTRACT_ADDRESS,
-        "Dummy Contract got deployed on unexpected address"
+    let contract = DummyCoreContract::deploy(&setup.provider).await.expect("Unable to deploy address");
+    let ethereum_settlement_client = EthereumSettlementClient::with_test_settings(
+        setup.provider.clone(),
+        Some(*contract.address()),
+        setup.rpc_url,
+        None,
```

Comment:
Why are we not sending the `DummyContractAddress` here ? and sending `None` instead.